src/plugins.c

```diff
@@ -2000,7 +2000,7 @@ gboolean geany_plugin_register_proxy(GeanyPlugin *plugin, const gchar **extensio
 	foreach_list(node, active_proxies.head)
 	{
 		proxy = node->data;
-		g_return_if_fail(p != proxy->plugin);
+		g_return_val_if_fail(p != proxy->plugin, FALSE);
```

Comment:
The diff doesn't show but there's a comment about that just above ;-)

/me is sitting on the airport at just after midnight waiting for the flight